src/verifiers/issuerIdentity/dnsTxt/openAttestationDnsTxt.ts

```diff
@@ -59,103 +57,141 @@ const resolveIssuerIdentity = async (
+            OpenAttestationDnsTxtCode[OpenAttestationDnsTxtCode.INVALID_ISSUERS]
+          );
+
+        if (!smartContractAddress)
+          throw new CodedError(
+            "Smart contract address not found in identity proof",
+            OpenAttestationDnsTxtCode.INVALID_ISSUERS,
+            OpenAttestationDnsTxtCode[OpenAttestationDnsTxtCode.INVALID_ISSUERS]
+          );
+        return resolveIssuerIdentity(location, smartContractAddress, options);
       }
-    },
-    {
+      const invalidResponse: Identity = {
+        status: "INVALID",
+        reason: {
+          message: "Issuer is not using DNS-TXT identityProof type",
+          code: OpenAttestationDnsTxtCode.INVALID_ISSUERS,
+          codeString: OpenAttestationDnsTxtCode[OpenAttestationDnsTxtCode.INVALID_ISSUERS],
+        },
+      };
+      return invalidResponse; // eslint is happy, so am I (https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/no-object-literal-type-assertion.md)
+    })
+  );
+
+  const invalidIdentity = identities.find((identity): identity is InvalidIdentity => identity.status !== "VALID");
+  if (invalidIdentity) {
+    return {
       name,
       type,
-      unexpectedErrorCode: OpenAttestationDnsTxtCode.UNEXPECTED_ERROR,
-      unexpectedErrorString: OpenAttestationDnsTxtCode[OpenAttestationDnsTxtCode.UNEXPECTED_ERROR],
-    }
-  ),
+      data: identities,
+      reason: invalidIdentity.reason,
+      status: "INVALID",
+    };
+  }
+  return {
+    name,
+    type,
+    data: identities,
+    status: "VALID",
+  };
+};
+
+const verifyV3 = async (document: v3.WrappedDocument, options: VerifierOptions): Promise<VerificationFragment> => {
+  if (
+    document.openAttestationMetadata.proof.method !== v3.Method.DocumentStore &&
+    document.openAttestationMetadata.proof.method !== v3.Method.TokenRegistry
+  )
+    throw new CodedError(
+      "DNS-TXT is only supported with documents issued using document store or token registry",
+      OpenAttestationDnsTxtCode.UNSUPPORTED,
+      OpenAttestationDnsTxtCode[OpenAttestationDnsTxtCode.UNSUPPORTED]
+    );
+  const smartContractAddress = document.openAttestationMetadata.proof.value;
+  const { identifier } = document.openAttestationMetadata.identityProof;
+  const issuerIdentity = await resolveIssuerIdentity(identifier, smartContractAddress, options);
+
+  return {
+    name,
+    type,
+    data: {
+      identifier: issuerIdentity.location,
+      value: issuerIdentity.value,
+    },
+    ...(issuerIdentity.status === "INVALID" ? { reason: issuerIdentity.reason } : {}),
+    status: issuerIdentity.status,
+  };
+};
+
+const verify: VerifierType["verify"] = withCodedErrorHandler(
+  async (document, options) => {
+    if (utils.isWrappedV2Document(document)) return verifyV2(document, options);
+    if (utils.isWrappedV3Document(document)) return verifyV3(document, options);
+    throw new CodedError(
+      "Unrecognized document",
```

Comment:
message is a bit vague, "Document does not match either v2 or v3 formats" ?

Reply:
done